examples/allgather/mpi_equivalent/mpiallgather.cpp

```diff
@@ -7,7 +7,19 @@
 #include <stdio.h>
 #include <time.h>
 #include <math.h>
+#if defined(__clang__)
+#  pragma clang diagnostic push
+#  pragma clang diagnostic ignored "-Wcast-qual"
+#elif defined (__GNUC__)
+#  pragma GCC diagnostic push
+#  pragma GCC diagnostic ignored "-Wcast-qual"
+#endif
 #include <mpi.h>
+#if defined(__clang__)
+#  pragma clang diagnostic pop
+#elif defined (__GNUC__)
+#  pragma GCC diagnostic pop
+#endif
 
```

Comment:
Should we wrap this into a separate header file instead of wrapping all `#include <mpi.h>` with this #pragma goo?

Reply:
Yes, that's a good idea. I've updated the PR with that.